python/cuml/dask/preprocessing/LabelEncoder.py

```diff
@@ -0,0 +1,194 @@
+
+        print(ord_label.compute())
+
+    Output:
+
+    .. code-block:: python
+
+        [0 1 2 0 0]
+
+        [0 1 2 0 0]
+
+           num_col cat_col  encoded
+        0       10       a        0
+        1       20       b        1
+        2       30       c        2
+        3       30       a        0
+        4       30       a        0
+
+        [2 0]
+
+        0    a
+        1    a
+        2    b
+        0    c
+        1    b
+        dtype: object
+
+    """
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(LabelEncoder, self).__init__(client=client,
+                                           verbose=verbose,
+                                           **kwargs)
+
+    def fit(self, y):
+        """
+        Fit a LabelEncoder (nvcategory) instance to a set of categories
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Series containing the categories to be encoded. Its elements
+            may or may not be unique
+
+        Returns
+        -------
+        self : LabelEncoder
+            A fitted instance of itself to allow method chaining
+
+        Notes
+        --------
+        Number of unique classes will be collected at the client. It'll
+        consume memory proportional to the number of unique classes.
+        """
+        _classes = y.unique().compute()
+        el = first(y) if isinstance(y, Sequence) else y
+        self.datatype = ('cudf' if isinstance(el, (dcDataFrame, daskSeries))
+                         else 'cupy')
+        self._set_internal_model(LE(**self.kwargs).fit(y, _classes=_classes))
+        return self
+
+    def fit_transform(self, y, delayed=True):
+        """
+        Simultaneously fit and transform an input
+
+        This is functionally equivalent to (but faster than)
+        LabelEncoder().fit(y).transform(y)
+        """
+        return self.fit(y).transform(y, delayed=delayed)
+
+    def transform(self, y, delayed=True):
+        """
+        Transform an input into its categorical keys.
+
+        This is intended for use with small inputs relative to the size of the
+        dataset. For fitting and transforming an entire dataset, prefer
+        `fit_transform`.
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Input keys to be transformed. Its values should match the
+            categories given to `fit`
+
+        Returns
+        -------
+        encoded : dask_cudf.Series
+            The ordinally encoded input series
+
+        Raises
+        ------
+        KeyError
+            if a category appears that was not seen in `fit`
+        """
+        if self._get_internal_model() is not None:
+            return self._transform(y,
+                                   delayed=delayed,
+                                   output_dtype='int64',
+                                   output_collection_type='cudf')
```

Comment:
Is there a way for a user to set the output collection type? Seems like it's always fixed to cudf. Do we need to add a cupy version as well? Seems like the sklearn equivalent returns an array so having both options could be desirable here.
Also, does the output_dtype have to be int64? Int32 seems adequate for the bulk of applications so we could make it configurable.

Reply:
Do you mean if we should have cupy option in `_transform`? OneHotEncoder uses cupy for output collection type.

Reply:
Yup, exactly. What do you think - would it make sense to have both options? I think Corey was agreeing with that approach too. Ok to file an issue for that and push this as is.

Reply:
I agree, I think it's good to have both approaches at least for now. We have to see if other files reference the cupy option too.

Reply:
Yeah, makes sense